### PR TITLE
add DODO to dex_aggregator

### PR DESCRIPTION
**Pull request by owen05**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Abnormal trades:
https://dune.com/queries/2506615?d=1

And this trade is unrelated to this PR.

cc. @jeff-dude

**Comment by jeff-dude** [comment]
> Abnormal trades: https://dune.com/queries/2506615?d=1
> 
> Etherscan: https://etherscan.io/tx/0x14613a8f439f5f65b3235fbae95b67bc5f6aa7b0937af73e8df593e00d8abb12
> 
> And this trade is unrelated to this PR.
> 
> cc. @jeff-dude

yes, this is known bug, listed here #3400

**Comment by jeff-dude** [comment]
for the sake of moving this PR along, i wonder if you can edit the gh workflow for the `dbt test` command to only be warning, rather than error? should be a setting you can apply in there. we'll just have to revert after.

File: models/dex/dex_schema.yml
Comment (Hosuke) [inline models/dex/dex_schema.yml:66]: I add a severity setting here.

Comment (jeff-dude) [reply models/dex/dex_schema.yml:66]: nice. i think we should revert, since we run this test in prod and want errors to appear and send alerts to us

**Review by jeff-dude — APPROVED** [review]
approving, but after revert we can explore merge

**Comment by Hosuke** [comment]
Hi @owen05 @jeff-dude 
There may be a merging sequence order needed for `DODO` related PRs.
Otherwise there may be conflicts like this: https://github.com/duneanalytics/spellbook/pull/3433#issuecomment-1565181044

We may firstly introduce an `dodo_aggregator.trades` roll-up spell, and remove the aggregator.trades from previous dex.trades roll-up spells. eg. https://github.com/duneanalytics/spellbook/blob/main/models/dodo/optimism/dodo_optimism_trades.sql#L16

@owen05 I suggest we make a clean-up PR for all these recent dodo trades updates with overall restructure.

**Comment by owen05** [comment]
I close this pr and reopen other one #3436 . And I combine DODO FeeProxy updates and DODO aggregator both into that new branch @Hosuke

**Pull request closed without merging**
